mis_builder/static/src/js/mis_report_widget.js

```diff
@@ -74,13 +101,104 @@ odoo.define('mis_builder.widget', function (require) {
+                context: context,
+            }).then(function (result) {
+                var record = result[0];
+                self.hide_analytic_filters = record['hide_analytic_filters'];
+            });
+
+            return $.when(this._super.apply(this, arguments), def1, def2, def3, def_hide_analytic_filters);
+        },
+
+        start: function () {
+            var self = this;
+            self._super.apply(self, arguments);
+            self.add_filters();
+        },
+
+        init_filter_value: function(field_object, attr_name) {
+            var self = this;
+            var filter = self.filter_values[attr_name];
+            if (filter !== undefined && filter['value'] !== undefined) {
+                field_object.value = filter['value'][0];
+                field_object.m2o_value = filter['value'][1];
+            }
+        },
+
+        init_filter: function(filter_name) {
+            var self = this;
+            if(self.filter_values[filter_name] === undefined) {
+                self.filter_values[filter_name] = {};
+            }
+        },
+
+        set_filter_value: function(field_object, attr_name, new_val) {
+            var self = this;
+            self.init_filter(attr_name);
+            self.filter_values[attr_name]['value'] =
+                [new_val.id, new_val.display_name] || undefined;
+        },
+
+        add_filters: function () {
+            var self = this;
+            if (self.hide_analytic_filters) {
+                return;
+            }
+            self.add_analytic_account_filter();
+        },
+
+        add_analytic_account_filter: function () {
+            var self = this;
+            if (!self.has_group_analytic_accounting) {
+                return;
+            }
+            if (self.analytic_account_id_m2o) {
+                // Prevent errors with autocomplete
+                self.analytic_account_id_m2o.destroy();
+            }
+            var field_name = 'analytic_account_id';
+            self.analytic_account_id_m2o = new relational_fields.FieldMany2One(self, "analytic_account_filter_id", self.record, {
+                mode: 'edit',
+                viewType: self.viewType,
+                attrs: {
+                    can_write: false,
+                    can_create: false,
+                },
+            });
+            self.init_filter_value(self.analytic_account_id_m2o, field_name);
+
+            // Add field to view
+            self.analytic_account_id_m2o.prependTo(self.get_mis_builder_filter_box());
+
+            self.analytic_account_id_m2o.$external_button.hide();
         },
 
         refresh: function () {
             this.replace();
         },
 
+        on_field_changed: function (event) {
+            var self = this;
+            if (event && event.data.changes) {
+                var changes = event.data.changes;
+                if (changes['analytic_account_filter_id']) {
```

Comment:
```suggestion
                if (changes['analytic_account_filter_id'] !== undefined) {
```
In case the value is "false"